spaceapi/src/sensors.rs

```diff
@@ -7,21 +7,67 @@ use utils::Optional;
 
 //--- Templates ---//
 
-/// An enum of all possible sensor templates.
+/// A trait for all possible sensor templates.
+/// the `SensorTemplate`s are capable of registering themselves in a `Sensors` struct
+pub trait SensorTemplate : Send+Sync {
+    fn to_sensor(&self, value_str: &str, sensors: &mut Sensors);
+}
+
+#[derive(Debug, Clone)]
+pub struct PeopleNowPresentSensorTemplate {
+    pub location: Optional<String>,
+    pub name: Optional<String>,
+    pub names: Optional<Vec<String>>,
+    pub description: Optional<String>,
+}
+
+impl SensorTemplate for PeopleNowPresentSensorTemplate {
+    fn to_sensor(&self, value_str: &str, sensors: &mut Sensors) {
+        if value_str.parse::<i64>().map(|value|{
+            let sensor = PeopleNowPresentSensor {
+                location: self.location.clone(),
+                name: self.name.clone(),
+                names: self.names.clone(),
+                description: self.description.clone(),
+                value: value,
+            };
+
+            match sensors.people_now_present {
+                Optional::Value(ref mut vec_sensors) => vec_sensors.push(sensor),
```

Comment:
Maybe put these instructions in curly braces, to show that it's not just a return value but an instruction with side effects?